compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt

```diff
@@ -115,12 +115,13 @@ inline fun IrProperty.addSetter(builder: IrFunctionBuilder.() -> Unit = {}): IrS
     }
 
 fun IrFunctionBuilder.buildFun(originalDescriptor: FunctionDescriptor? = null): IrFunctionImpl {
-    val wrappedDescriptor = if (originalDescriptor is DescriptorWithContainerSource)
-        WrappedFunctionDescriptorWithContainerSource(originalDescriptor.containerSource)
-    else if (originalDescriptor != null)
-        WrappedSimpleFunctionDescriptor(originalDescriptor)
-    else
-        WrappedSimpleFunctionDescriptor()
+    val wrappedDescriptor = when(originalDescriptor) {
```

Comment:
More of a side note, but I wonder if we can make this more general and implement `DescriptorsToIrRemapper.remapDeclaredSimpleFunction` this way and use that method everywhere `WrappedFunctionDescriptorWithContainerSource` is currently instantiated? Because I suspect other call sites might also need special descriptor instances for property accessors.

Also, maybe having `WrappedProperty{G,S}etterDescriptorWithContainerSource` would be useful.

But since such change might break something else, it's not directly related to this one.

Reply:
I wasn't aware of this code, but yes, we should probably produce `WrappedFunctionDescriptorWithContainerSource` here as well. This doesn't seem to be needed for property accessors, though, since these are already a special case in the inliner. I'm honestly not sure *why* the inliner has several different ways of identifying external declarations, so I don't know if we wouldn't end up breaking something else by introducing a new `...DescriptorWithContainerSource` variant.